src/kit.ts

```diff
@@ -1163,48 +1164,52 @@ export async function getVSKitEnvironment(kit: Kit): Promise<Environment | null>
     return varsForVSInstallation(requested, kit.visualStudioArchitecture!, kit.preferredGenerator?.platform);
 }
 
+/**
+ * kit.environmentVariables have higher priority, we expand `Environment` with
+ * `environmentSetupScript` first, then expand and update `Environment` with`environmentVariables`
+ * @param kit The kit for evaluate `Environment`
+ * @param opts The expand options for evaluate `Environment`
+ * @returns `Environment`
+ */
 export async function effectiveKitEnvironment(kit: Kit, opts?: expand.ExpansionOptions): Promise<Environment> {
-    let host_env: Environment | undefined;
-    const kit_env = EnvironmentUtils.create(kit.environmentVariables);
-    if (opts) {
-        for (const env_var of Object.keys(kit_env)) {
-            kit_env[env_var] = await expand.expandString(kit_env[env_var], opts);
-        }
-    }
+    let host_env: Environment = process.env;
     if (kit.environmentSetupScript) {
         const shell_vars = await getShellScriptEnvironment(kit, opts);
         if (shell_vars) {
             host_env = shell_vars;
         }
     }
-    if (host_env === undefined) {
-        // get host_env from process if it was not set by shell script before
-        host_env = process.env;
-    }
-    if (kit.visualStudio && kit.visualStudioArchitecture) {
-        const vs_vars = await getVSKitEnvironment(kit);
-        if (vs_vars) {
-            return vs_vars;
-        }
+    let env = EnvironmentUtils.create(host_env);
+    const kit_env = EnvironmentUtils.create(kit.environmentVariables);
+    for (const env_var of Object.keys(kit_env)) {
+        const expandOptions: ExpansionOptions = {
```

Comment:
`expandOptions` doesn't appear to change from iteration to iteration.  Can you promote it out of the for-loop?